src/doc/style-guide/src/items.md

```diff
@@ -450,7 +450,7 @@ entries, format it across multiple lines as with a type alias.
 
 When writing extern items (such as `extern "C" fn`), always specify the ABI.
 For example, write `extern "C" fn foo ...`, not `extern fn foo ...`, or
-`extern "C" { ... }`.
+`unsafe extern "C" { ... }`.
```

Comment:
I actually feel like this is kinda vague wording from the style guide itself. As far as I can tell, it's encouraging the user to write:

```
extern "C" fn foo() {}
```

Instead of:

```
extern "C" {
    fn foo();
}
```

Right? I guess whether or not we put `unsafe` on this block is... kinda pointless, IMO?

Reply:
The existing wording in the style guide is a bit broken in that it seems to equivocate between `extern { .. }` and `extern fn`.

Reply:
It's the first version of @traviscross what the paragraph is talking about as it is talking about specifying the ABI. You can do it with `extern "C" fn foo` or with `unsafe extern "C"` which is preferred now with unsafe extern blocks over `extern "C"`.
I agree that the wording may not be the best.
I can try another version to clarify what is said.